project/BuildSettings.scala

```diff
@@ -0,0 +1,87 @@
+// SBT
+import sbt._
+import Keys._
+
+// sbt-assembly
+import sbtassembly._
+import sbtassembly.AssemblyKeys._
+
+/**
+  * Common settings-patterns for Snowplow apps and libraries.
+  * To enable any of these you need to explicitly add Settings value to build.sbt
+  */
+object BuildSettings {
+
+  lazy val buildSettings = Seq[Setting[_]](
+    scalacOptions := Seq(
+      "-deprecation",
+      "-encoding", "UTF-8",
+      "-feature",
+      "-unchecked",
+      "-Ywarn-dead-code",
+      "-Ywarn-inaccessible",
+      "-Ywarn-infer-any",
+      "-Ywarn-nullary-override",
+      "-Ywarn-nullary-unit",
+      "-Ywarn-numeric-widen",
+      "-language:higherKinds",
+      "-Ywarn-unused",
+      "-Ypartial-unification",
+      "-Ywarn-value-discard"
+    ),
+    javacOptions := Seq(
+      "-source", "1.8",
+      "-target", "1.8",
+      "-Xlint"
+    )
+  )
+
+  // Makes package (build) metadata available withing source code
+  lazy val scalifySettings = Seq(
```

Comment:
This is just a dirty hack for us to have name of project **in code** automatically. You bump version in definition - it bumps in program. Not important and I didn't see any projects apart from ours using this practive.